include/pybind11/detail/class.h

```diff
@@ -302,6 +307,8 @@ extern "C" inline PyObject *pybind11_object_new(PyTypeObject *type, PyObject *,
 /// An `__init__` function constructs the C++ object. Users should provide at least one
 /// of these using `py::init` or directly with `.def(__init__, ...)`. Otherwise, the
 /// following default function will be used which simply throws an exception.
+///
+/// Use `get_internals().default_object_init()` instead of using this function directly
 extern "C" inline int pybind11_object_init(PyObject *self, PyObject *, PyObject *) {
```

Comment:
This function is used in two places in this file. You caught the one on 623, but not the one on line 422. Either the added comment is misleading or we need to fix line 422.